yt_dlp/extractor/afreecatv.py

```diff
@@ -253,6 +253,44 @@ def _real_extract(self, url):
         return self.playlist_result(entries, video_id, multi_video=True, **common_info)
 
 
+class AfreecaTVCatchStoryIE(AfreecaTVBaseIE):
+    IE_NAME = 'afreecatv:catchstory'
+    IE_DESC = 'afreecatv.com catch story'
+    _VALID_URL = r'https?://vod\.afreecatv\.com/player/(?P<id>\d+)/catchstory'
+    _TESTS = [{
+        'url': 'https://vod.afreecatv.com/player/103247/catchstory',
+        'info_dict': {
+            'id': '103247',
+        },
+        'playlist_count': 2,
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        data = self._download_json(
+            'https://api.m.afreecatv.com/catchstory/a/view', video_id, headers={'Referer': url},
+            query={'aStoryListIdx': '', 'nStoryIdx': video_id}, impersonate=True)
+
+        return self.playlist_result(self._entries(data), video_id)
+
+    @staticmethod
+    def _entries(data):
+        # 'files' is always a list with 1 element
+        yield from traverse_obj(data, (
+            'data', lambda _, v: v['story_type'] == 'catch',
+            'catch_list', lambda _, v: v['files'][0]['file'], {
+                'id': ('files', 0, 'file_info_key', {str}),
+                'url': ('files', 0, 'file', {url_or_none}),
+                'ext': ('src_movie', {lambda v: determine_ext(v, 'mp4')}),
```

Comment:
we don't want this. the test's url is an m3u8 and this value is mp4. it will not download correctly if yt-dlp thinks an m3u8 is an mp4
```suggestion
```

Reply:
I don't understand why, but sometimes it has the m3u8 extension but is actually mp4. Since this problem only occurs in a very few cases, I will revert that commit.

Reply:
No, it's an m3u8, and yt-dlp downloads it with its HLS downloader and muxes it to mp4

Reply:
if there are other catchstory links where the format is actually an mp4 could we add a test for one of those? btw do these stories expire?

Reply:
Because it is a similar service to YouTube's Shorts, it does not seem to be deleted. But I don't think there's a need to deal with edge cases.
In this case, it was uploaded from a dummy account. I think it was uploaded for development testing.

Reply:
> But I don't think there's a need to deal with edge cases.

If you have a link where the format url is a direct mp4 then we should add it; test coverage is good